purchases/src/main/kotlin/com/revenuecat/purchases/CustomerInfoHelper.kt

```diff
@@ -82,6 +93,116 @@ internal class CustomerInfoHelper(
+                            put("request_date", formattedDate)
+                            put("request_date_ms", requestDate.time)
+                            put("subscriber", JSONObject().apply {
+                                put("original_app_user_id", appUserID)
+                                // put("original_application_version", "1.0")
+                                put("entitlements", generateEntitlementsResponse(cachedEntitlements, allPurchases,
+                                    activePurchasesByHashedToken))
+                                put("first_seen", formattedDate)
+                                put("original_purchase_date", formattedDate)
+                                put("non_subscriptions", JSONObject()) // TODO
+                                put("subscriptions", generateSubscriptions(allPurchases, activePurchasesByHashedToken))
+                                put("management_url", "https://play.google.com/store/account/subscriptions")
+                            })
+                        }
+
+                        val customerInfo = CustomerInfoFactory.buildCustomerInfo(
+                            jsonObject,
+                            requestDate,
+                            VerificationResult.NOT_REQUESTED
+                        )
+                        onSuccess?.invoke(storeTransaction, customerInfo)
+                    },
+                    onError = { error ->
+                        log(LogIntent.GOOGLE_ERROR, error.message)
+                    }
+                )
+            },
+            onReceivePurchaseHistoryError = {
+            }
+        )
+    }
+
+    private fun generateSubscriptions(
+        allPurchases: List<StoreTransaction>,
+        activePurchasesByHashedToken: Map<String, StoreTransaction>
+    ): JSONObject {
+        val subscriptions = JSONObject()
+        val tomorrow = getTomorrowDate()
+
+        // TODO: allPurchases is actually latest purchase for each product
+        allPurchases.filter { it.type == ProductType.SUBS }.forEach {
+            subscriptions.put(it.skus[0], JSONObject().apply {
+                put("billing_issues_detected_at", JSONObject.NULL)
+                put("is_sandbox", false)
+                put("original_purchase_date", false)
+                val purchaseDate = Iso8601Utils.format(Date(it.purchaseTime))
+                put("original_purchase_date", purchaseDate)
+                put("purchase_date", purchaseDate)
+                put("store", "play_store")
+                put("unsubscribe_detected_at", JSONObject.NULL)
+                put("expires_date", Iso8601Utils.format(tomorrow)) // TODO: tomorrow only if it's active
+                put("period_type", "normal") // TODO
+            })
+        }
+        return subscriptions
+    }
+
+    private fun generateEntitlementsResponse(
+        productEntitlementMapping: Map<String, String>,
```

Comment:
This will be a `Map<String, List<String>>` or a `ProductEntitlementMapping` object (#888) either should work